docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Elytron_OIDC_Client.adoc

```diff
@@ -173,6 +173,195 @@ It is also possible to configure the scope values that would be used via the sco
+plaintext. When it is plaintext, this is indicated by use of the `none` algorithm in the JWS Header. The signature
+algorithm for the request object can be specified using the `request-object-signing-algorithm` attribute. This attribute
+is of type string and the default value is `none`, indicating that the request object will be sent as plaintext.
+
+If any asymmetric signing algorithm is specified as the signing algorithm for the request object, then the client keys
+will also need to be configured. This includes any RSA or Elliptical Curve algorithms. This is done using the following
+attributes:
+
+* `request-object-signing-keystore-file`: The absolute path to the client keystore file.
+* `request-object-signing-keystore-password`: The password to access the keystore file.
+* `request-object-signing-key-password`: The password to access the key within the keystore file.
+* `request-object-signing-key-alias`: The alias by which the key identified with within the keystore file.
+* `request-object-signing-keystore-type`: The keystore type for the file. This can be either "PKCS12" or "JKS".
+
+For `none` algorithm, we are not signing the JWT, so we don't need to configure a keystore. Similarly, for any HMAC
+algorithms, we are using the client secret to sign the JWT. Hence, we don't need to configure a keystore.
+
+NOTE: Although the default value for the request-object-signing-algorithm is `none`, indicating an unsigned JWT, this may not
+be allowed by all OpenID providers. Please refer to the OpenID metadata, specifically the `request_object_signing_alg_values_supported`
+section to determine which algorithms are supported.
+
+The keystore file can be created using `keytool` on the terminal using the following commands:
+```
+keytool -genkeypair -alias <keystore_alias> -keyalg <algorithm> -keysize <key_size> -validity <validity_in_days>
+-keystore <keystore_name> -dname "<distinguished_name>" -keypass <private_key_password> -storepass <keystore_password>
+```
+An example for this is:
+```
+$ keytool -genkeypair -alias exampleClientKeyStore -keyalg RSA -keysize 2048 -validity 365 -keystore
+exampleclient.keystore.pkcs12 -dname "CN=client" -keypass secret -storepass secret
+```
+
+Alternatively, the keystore file can also be created using the
+https://docs.wildfly.org/30/wildscribe/subsystem/elytron/key-store/index.html[Elytron subsystem].
+
+
+NOTE: For elliptical curve keys, if you are using a keystore of type JKS, then keysize may not matter. However, if you are
+using PKCS12, you cannot specify keysize using the `keysize` argument. Instead, you must use `-groupname` and set it to
+"secp256r1" if you are planning on using the "ES256" algorithm, "secp384r1" for "ES384" and so on. However, for RSA keys,
+keys of size 2048 would be applicable for the supported RSXXX and PSXXX algorithms regardless of the SHA value.
+
+==== Configuring Client Keys for Keycloak
```

Comment:
Just wondering, do we need this section here or can it be covered in the corresponding use-case driven guide?